### ISSUE TITLE
low performance on test set

### ISSUE THREAD

**Issue by patcharees** [RUCAIBox/RecBole#1394]
I used FPMC with the following config. The performance of validation set is like double of test set (overfitting). Any suggestions how should I adjust the config/training parameters?

*** config ****
Training Hyper Parameters:
epochs = 300
train_batch_size = 2048
learner = adam
learning_rate = 0.001
neg_sampling = {'uniform': 2}
eval_step = 1
stopping_step = 10
clip_grad_norm = None
weight_decay = 0.0
loss_decimal_place = 4

Evaluation Hyper Parameters:
eval_args = {'split': {'LS': 'valid_and_test'}, 'order': 'TO', 'mode': 'full', 'group_by': 'user'}
repeatable = True
metrics = ['Recall', 'MRR', 'NDCG', 'Hit', 'Precision']
topk = [4]
valid_metric = Hit@4
valid_metric_bigger = True
eval_batch_size = 4096
metric_decimal_place = 4

Dataset Hyper Parameters:
field_separator =
seq_separator =
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = label
threshold = None
NEG_PREFIX = neg_
load_col = {'inter': ['user_id', 'item_id', 'rating', 'timestamp']}
unload_col = None
unused_col = None
additional_feat_suffix = None
rm_dup_inter = None
val_interval = None
filter_inter_by_user_or_item = True
user_inter_num_interval = [0,inf)
item_inter_num_interval = [0,inf)
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = None
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 50
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id
benchmark_filename = None

Other Hyper Parameters:
wandb_project = recbole
require_pow = False
embedding_size = 256
MODEL_TYPE = ModelType.SEQUENTIAL
MODEL_INPUT_TYPE = InputType.PAIRWISE
eval_type = EvaluatorType.RANKING
device = cpu
train_neg_sample_args = {'strategy': 'by', 'by': 2, 'distribution': 'uniform', 'dynamic': 'none'}
eval_neg_sample_args = {'strategy': 'full', 'distribution': 'uniform'}

17 Aug 14:55    INFO  grieg_wo_weekday_2022-06-01
The number of users: 40
Average actions of users: 208.2051282051282
The number of items: 318
Average actions of items: 25.615141955835963
The number of inters: 8120
The sparsity of the dataset: 36.16352201257862%
Remain Fields: ['user_id', 'item_id', 'rating', 'timestamp']
17 Aug 14:55    INFO  [Training]: train_batch_size = [2048] negative sampling: [{'uniform': 2}]
17 Aug 14:55    INFO  [Evaluation]: eval_batch_size = [4096] eval_args: [{'split': {'LS': 'valid_and_test'}, 'order': 'TO', 'mode': 'full', 'group_by': 'user'}]
17 Aug 14:55    INFO  FPMC(
  (UI_emb): Embedding(40, 256)
  (IU_emb): Embedding(318, 256)
  (LI_emb): Embedding(318, 256, padding_idx=0)
Trainable parameters: 254464

**** Metric for the validation and test set ***
best valid : OrderedDict([('recall@4', 0.4444), ('mrr@4', 0.3796), ('ndcg@4', 0.396), ('hit@4', 0.4444), ('precision@4', 0.1111)])
test result: OrderedDict([('recall@4', 0.1852), ('mrr@4', 0.1142), ('ndcg@4', 0.1319), ('hit@4', 0.1852), ('precision@4', 0.0463)])

**Comment by zhengbw0324** [comment]
@patcharees Hello, thanks for your attention to RecBole!
We have checked your configuration file and found that the data split method you use is "leave one method",but your data quantity is very small, only 40 users and 318 items. Therefore, the distribution of the divided verification set and test set may be quite different.

**Comment by patcharees** [comment]
@zhengbw0324 Thank you for comment!

Unfortunately I cannot have more data. So I tried simple model like FPMC and FOSSIL with low embedding size. Do you have any suggestions what I should try with this small data? Or other models?

**Comment by zhengbw0324** [comment]
@patcharees
You can try using the ratio-based splitting instead of the leave-one-out splitting.
For example:
```yaml
eval_args:                      
  split: {'RS':[0.8,0.1,0.1]}   
  group_by: user       
  order: TO 
  mode: full
```
And you can experiment multiple times or do a hyperparameter search using `run_hyper.py`.

**Comment by patcharees** [comment]
Unfortunately it is not better. Any others parameter I should try?

For very small dataset, is there any model I should try in addition to FOSSIL and FPMC?

**Comment by zhengbw0324** [comment]
For the introduction of all models included in RecBole, please refer to the [documentation](https://recbole.io/docs/user_guide/model_intro.html). If there are problems with multiple models, it indicates that there is a problem with your dataset or configuration. 

You can try to adjust the dataset filtering strategy, negative sampling strategy or dataset splitting strategy. For example, use `val_interval` to filter data with a small rating value. The specific usage method and other dataset related configuration parameters are [here](https://recbole.io/docs/user_guide/config/data_settings.html).
